mono/metadata/w32file.c

```diff
@@ -492,13 +492,13 @@ ves_icall_System_IO_MonoIO_FindNextFile (HANDLE hnd, MonoString **file_name, gin
 	res = mono_w32file_find_next (hnd, &data);
 
 	if (res == FALSE) {
-		*file_name = NULL;
+		MONO_HANDLE_ASSIGN (file_name, NULL_HANDLE_STRING);
 		*file_attr = 0;
 		*ioerror = mono_w32error_get_last ();
 		return res;
 	}
 
-	mono_gc_wbarrier_generic_store (file_name, (MonoObject*) mono_string_from_utf16_checked (data.cFileName, &error));
+	MONO_HANDLE_ASSIGN (file_name, (MonoObject*) mono_string_from_utf16_checked (data.cFileName, &error));
```

Comment:
MONO_HANDLE_ASSIGN takes two handles, not a `MonoObject*`.  You want `mono_string_new_utf16_handle`